src/main/java/org/apache/commons/text/cases/CharacterDelimitedCase.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.text.cases;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.CharUtils;
+
+/**
+ * DelimitedCase is a case in which the true alphabetic case of the characters is ignored by default
+ * and tokens themselves are determined by the presence of a delimiter between each token.
+ */
+public class CharacterDelimitedCase implements Case {
+
+    /** Delimiters to be used when parsing. */
+    private List<Integer> parseDelimiters;
+
+    /** Delimiter to be used when formatting. */
+    private String formatDelimiter;
+
+    /**
+     * Constructs a new Delimited Case with null delimiters.
+     */
+    CharacterDelimitedCase() {
+        this(null, null);
+    }
+
+    /**
+     * Constructs a new Delimited Case.
+     *
+     * @param delimiter the character to use as both the parse and format delimiter
+     */
+    CharacterDelimitedCase(char delimiter) {
+        this(new char[] { delimiter }, CharUtils.toString(delimiter));
+    }
+
+    /**
+     * Constructs a new delimited case.
+     *
+     * @param parseDelimiters the array of delimiters to use when parsing
+     * @param formatDelimiter the delimiter to use when formatting
+     */
+    CharacterDelimitedCase(char[] parseDelimiters, String formatDelimiter) {
+        super();
+        this.parseDelimiters = generateDelimiterList(parseDelimiters);
+        this.formatDelimiter = formatDelimiter;
+    }
+
+    /**
+     * Formats tokens into Delimited Case.
+     * <p>
+     * Tokens are iterated on and appended to an output stream, with an instance of a
```

Comment:
"iterated on" sounds off; maybe just drop it

delete instance of. The delimiter might not even be an object. I guess it is a string but then it might not be a single character